### PR TITLE
Change: Forbid dock placement on docking area

### PR DESCRIPTION
**Pull request by SamuXarick**
Negates dock placement if it would be blocking ship access to the docking location of another dock or of an oil rig station.

https://www.tt-forums.net/viewtopic.php?p=1176272

### CODE REVIEW

File: src/station_cmd.cpp [hunk 1]
```diff
@@ -2449,6 +2449,53 @@ bool HasStationInUse(StationID station, bool include_company, CompanyID company)
...
+					Axis axis_tc = DiagDirToAxis(rot);
+					if (axis != axis_tc) {
+						return_cmd_error(STR_ERROR_TOO_CLOSE_TO_ANOTHER_DOCK);
+					}
+				}
+			}
+		}
+
+		if (axis != AXIS_X && DiagDirToAxis(rot) == AXIS_X) {
+			if (IsTileType(tc + diff, MP_INDUSTRY) && GetIndustryGfx(tc + diff) == GFX_OILRIG_1 &&
```
Comment (nielsmh) [inline src/station_cmd.cpp:2488]: Will this work properly with NewGRF industries built on water with oil rig-style stations? (Such as fishing grounds in FIRS.)

Comment (SamuXarick) [reply src/station_cmd.cpp:2488]: Probably yes.

I copied from these rules:
https://github.com/OpenTTD/OpenTTD/blob/b3b89257f745481db211fbc0bb185660528ec00c/src/industry_cmd.cpp#L737-L750


If FIRS is following these rules to create a station at the fishing grounds, then the docking tile should be in the same location as that of the oil rig.

Comment (nielsmh) [reply src/station_cmd.cpp:2488]: Yeah, checked, FIRS does use an "Oil Rig" tile which is exactly the `GFX_OILRIG_1` tile ID for both Fishing grounds and Dredring site industries. It should work then.

**Review by nielsmh — DISMISSED** [review]

**Comment by J0anJosep** [comment]
I am not keen on these types of checks. If we had to check for any action whether it will block any vehicle or any station, the code would be much larger than it is now.

**Comment by TrueBrain** [comment]
We recently switched from Jenkins as CI to Azure Pipelines as CI. This means you need to rebase before this Pull Request will pass its checks. Sorry for the troubles!

**Comment by andythenorth** [comment]
This accounts for Oil Rigs, does it account for Newgrf industries built on water?  L2472 does that?

**Comment by SamuXarick** [comment]
>This accounts for Oil Rigs, does it account for Newgrf industries built on water?  L2472 does that?

I tried to comply with the rules found somewhere on industry_cmd.cpp

**Comment by andythenorth** [comment]
On reflection it's not essential to provide these kind of checks.  The benefit is too little for the complexity introduced, and the possibility of unintended side effects.  This PR is not likely to be accepted, so I'm closing it to keep things tidy.  Thanks for contributing!

**Pull request closed without merging**